react-ui/src/actions/detail.js

```diff
@@ -12,26 +12,31 @@ export const detailLoading = bool => {
   };
 };
 
-export const updateDetail = (location, happyhours = null) => {
+export const updateDetail = location => {
   return {
     type: 'UPDATE_DETAIL',
-    location,
-    happyhours
+    location
   };
 };
 
-export const fetchDetail = (id) => {
+export const fetchDetail = id => {
   return dispatch => {
     dispatch(detailLoading(true));
 
-    fetch(`/api/v1/locations/${id}/happyhours`)
+    const happyHours = fetch(`/api/v1/locations/${id}/happyhours`)
       .then(res => {
         dispatch(detailLoading(false));
         return res.json();
       })
-      .then(({ data }) => {
-        dispatch(updateDetail(location, data));
-      })
+      .then(({ data }) => data)
       .catch(error => dispatch(detailError(error)));
-  }
-}
+
+    const location = fetch(`/api/v1/locations/${id}`)
+      .then(res => res.json())
+      .then(({ data }) => data[0]);
+
+    Promise.all([happyHours, location]).then(data => {
+      dispatch(updateDetail({ data }));
```

Comment:
This looks a little strange. Data is an array then you're assigning it to an object with a  key of data. Seems unnecessary but I could be wrong.

Reply:
@coleworsley 
Good eye, Justyna just pulled that out and we updated DetailsPage.js.